infinitered/ReactNativeEssentials
# Local Notifications via Notifee

### Summary

This PR adds local notifications utilizing the [Notifee library](https://notifee.app/). Two types of mechanisms are used to trigger notifications: displaying notifications while app is in foreground and scheduling notifications for when app is in background.

- Adds notifee dependency.
- Adds a `notifications.ts` service which has helper functions and hooks for handling notifications.
- Adds logic for triggering local notifications when app is in foreground.
- Adds a scheduled notification trigger after apps goes to background.
- Adds miscellaneous helpers to `./shared`.

### How to verify

1. `yarn`
2. `yarn start`
3. `i` & `a`
4. Accept all permissions and additional settings (if prompted).
5. Test foreground notifications.
   1. Go to a game that has 0 reviews. 
   2. Mark it as a "favorite".
   3. Observe a notification.
   4. Click notification.
   5. Observe the "Review" screen for the notified game.
6. Test background notifications
   1. If you have a favorited game with 0 reviews, skip to step 4.
   2. Go to a game that has 0 reviews.
   3. Mark it as a "favorite" and dismiss the initial notification.
   4. Background the app and wait 5 seconds.
   5. Observe a notification.
   6. Click notification.
   7. Observe the "Review" screen for the notified game.
   8. Background the app and wait 5 seconds.
   9. Observe a notification.
   10. (iOS) Pull down notification and type review. (Android 13) Click "Review Game" in the notification and type review.
   11. Foreground app.
   12. Observe the review you added should be in the list. 

### Proofs







| iOS                                             | Android                                             |
| ----------------------------------------------- | --------------------------------------------------- |
| <video  src="https://github.com/infinitered/ReactNativeEssentials/assets/1775841/50d5aefb-7267-4a64-9682-5ec3dbf2f7db" /> | <video src="https://github.com/infinitered/ReactNativeEssentials/assets/1775841/66022e2d-5672-43d4-8b25-52138d4e7a70" /> |
